realm/app-manager/src/app.rs

```diff
@@ -0,0 +1,269 @@
+use std::os::linux::fs::MetadataExt;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use log::info;
+use nix::libc::{major, makedev, minor, S_IFBLK};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use uuid::Uuid;
+use warden_realm::ApplicationInfo;
+
+use super::Result;
+use crate::dm::crypt::{CryptDevice, CryptoParams, DmCryptTable, Key};
+use crate::dm::device::{DeviceHandleWrapper, DeviceHandleWrapperExt};
+use crate::dm::DeviceMapper;
+use crate::key::ring::KernelKeyring;
+use crate::key::KeySealing;
+use crate::launcher::{ApplicationHandler, Launcher};
+use crate::util::disk::read_device_size;
+use crate::util::fs::{
+    dirname, formatfs, mkdirp, mknod, mount, mount_overlayfs, read_to_string, stat, umount,
+    write_to_string, Filesystem,
+};
+use crate::util::serde::{json_dump, json_load};
+
+#[derive(Debug, Error)]
+pub enum ApplicationError {
+    #[error("Partition not found")]
+    PartitionNotFound(),
+}
```

Comment:
Minor: Maybe we could add information about partition (Uuid) that can't be found?

Reply:
done